### PR TITLE
feat(timepicker): added test plan

### PR DESCRIPTION
**Pull request by artemvfrolov**

### CODE REVIEW

File: src/spec/timepicker/timepicker.utils.spec.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    hour: hours || null,
+    minute: minutes || null,
+    seconds: second || null,
+    isPM: PM || null
+  };
+  return time;
+}
+
+describe('Runtime coverage. Utils: Timepicker', () => {
+  it('should is no value', () => {
```
Comment (SergeyKuryatnick) [inline src/spec/timepicker/timepicker.utils.spec.ts]: what?

File: src/spec/timepicker/timepicker.component.spec.ts [hunk 2]
```diff
@@ -1,175 +1,1030 @@
...
+
+function getElements(fixture: any, selector: string) {
+  return fixture.nativeElement.querySelectorAll(selector) as HTMLElement;
+}
+
+function getDebugElements(fixture: any, selector: string) {
+  return fixture.debugElement.queryAll(By.css(selector));
+}
+
+function testTime(hours?: number, minutes?: number, seconds?: number) {
```
Comment (SergeyKuryatnick) [inline src/spec/timepicker/timepicker.component.spec.ts:24]: You use this function in 3 specs - just move it to helper

File: src/spec/timepicker/timepicker.component.spec.ts [hunk 3]
```diff
@@ -1,175 +1,1030 @@
...
-  // });
-  //
-  // it('should canBeChanged true', () => {
-  //   context.readonlyInput = true;
-  //   context.canBeChanged();
-  // });
-
-  it('should changeHours', () => {
-    context.changeHours(3);
+    // поле часы и минуты отображаются
```
Comment (SergeyKuryatnick) [inline src/spec/timepicker/timepicker.component.spec.ts:71]: Maybe all comments should be on english?

File: src/spec/timepicker/timepicker.component.spec.ts [hunk 4]
```diff
@@ -1,175 +1,1030 @@
...
-  it('should changeMinutes canDecrementHours', () => {
-    context.canDecrementMinutes = false;
-    context.changeMinutes(-3);
-  });
+      expect(inputMinutes.value).toEqual('12');
+    });
+    // проверить данные в поле минуты корректные данные с неполным значением
+    it('should validate the data in the minutes input with valid data with half value', () => {
+      component.writeValue(testTime(0,2,0));
+      fixture.detectChanges();
```
Comment (SergeyKuryatnick) [inline src/spec/timepicker/timepicker.component.spec.ts:119]: Unneeded command in this test

**Review by SergeyKuryatnick — CHANGES_REQUESTED** [review]

**Comment by SergeyKuryatnick** [comment]
Try not to use `fixture.detectChanges()` in all tests, only if it necessary  :)

File: src/spec/timepicker/timepicker.component.spec.ts (hunk 3 above)
Comment (valorkin) [reply src/spec/timepicker/timepicker.component.spec.ts:71]: it is for me, so I can check test case translations

**Comment by SergeyKuryatnick** [comment]
Tests are failed. recheck pls.

**Review by SergeyKuryatnick — APPROVED** [review]

**Pull request merged**